google/resource_compute_instance.go

```diff
@@ -741,6 +753,54 @@ func expandComputeInstance(project string, d *schema.ResourceData, config *Confi
 	}, nil
 }
 
+func getAllStatusExcept(exceptedStatus string) []string {
+	allStatus := []string{
+		"PROVISIONING",
+		"REPAIRING",
+		"RUNNING",
+		"STAGING",
+		"STOPPED",
+		"STOPPING",
+		"SUSPENDED",
+		"SUSPENDING",
+		"TERMINATED",
+	}
+	for i, status := range allStatus {
+		if exceptedStatus == status {
+			return append(allStatus[:i], allStatus[i+1:]...)
+		}
+	}
+	return nil
+}
+
+func waitUntilInstanceHaveTheExpectedStatus(config *Config, d *schema.ResourceData) error {
```

Comment:
I used `d` as parameter here because in the `StateRefreshFunc` function, there is a call to `getInstance(config *Config, d *schema.ResourceData)` to retrieve the current instance (and its status). I believe I can create my own function `getInstanceStatus(name, region, project string) (status string)` but I felt like reusing the existing would be a good idea. Am I wrong?

Reply:
Good point, this approach is :+1:

Since `d` is mutable I generally try not to pass it too far down into functions for the sake of clarity when reading the code, but this file has a precedent of passing it down.